src/condor_utils/write_user_log.cpp

```diff
@@ -166,7 +167,7 @@ WriteUserLog::initialize(const ClassAd &job_ad, bool init_user)
 		job_ad.LookupString(ATTR_NT_DOMAIN, domain);
 
 		uninit_user_ids();
-		if ( ! init_user_ids(owner.c_str(), domain.c_str()) ) {
+		if ( ! init_user_ids_from_ad(job_ad) ) {
```

Comment:
I think init_user_ids_from_ad() needs flags that allow callers to specify that it is ok to fall back to using Owner and NTDomain when OS user attribute is missing.  These flags should only be set for users of the function that are outside the AP.